erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -618,13 +618,16 @@ def update_component_row(self, component_data, amount, component_type, additiona
 
 			component_row = self.append(component_type)
 			for attr in (
-				'depends_on_payment_days', 'salary_component', 'abbr'
```

Comment:
Okay, turns out that setting the `abbr` separately was never the issue. Python did not find a comma after `abbr` so it assumed `abbrdo_not_include_in_total` to be one string.

Reply:
It's my fault :sweat_smile: 
https://github.com/frappe/erpnext/pull/24824/files#diff-a6a219b6ecafbe24da6ed9f0453f790941824d8728616368a238a74e4a110a14R620

Reply:
![image](https://user-images.githubusercontent.com/16315650/120023543-93b1ac00-c00b-11eb-80c4-4803957b01b5.png)

Reply:
wait, implicit string concatenation this should've been caught by semgrep 🤔

Reply:
Was aware of the missing comma while raising the earlier fix but the deduction component was still not getting computed in some cases. I checked all occurrences of abbr. The salary_component_abbr field from Salary Component gets fetched with the alias abbr everywhere. So the deductions are coming through just fine now.